homeassistant/components/simplisafe/__init__.py

```diff
@@ -175,21 +186,141 @@ def _async_save_refresh_token(hass, config_entry, token):
+        system = systems[call.data[ATTR_SYSTEM_ID]]
+        await system.set_alarm_duration(call.data[ATTR_SECONDS])
+
+    @verify_system_exists
+    @v3_only
+    @_verify_domain_control
+    async def set_alarm_volume(call):
+        """Set the volume of a running alarm."""
+        await set_volume_parameter(call, "set_alarm_volume")
+
+    @verify_system_exists
+    @v3_only
+    @_verify_domain_control
+    async def set_chime_volume(call):
+        """Set the volume of the door chime."""
+        await set_volume_parameter(call, "set_chime_volume")
+
+    @verify_system_exists
+    @v3_only
+    @_verify_domain_control
+    async def set_entry_delay_away(call):
+        """Set the entry delay duration ("away" mode)."""
+        system = systems[call.data[ATTR_SYSTEM_ID]]
+        await system.set_entry_delay_away(call.data[ATTR_SECONDS])
+
+    @verify_system_exists
+    @v3_only
+    @_verify_domain_control
+    async def set_entry_delay_home(call):
+        """Set the entry delay duration ("home" mode)."""
+        system = systems[call.data[ATTR_SYSTEM_ID]]
+        await system.set_entry_delay_home(call.data[ATTR_SECONDS])
+
+    @verify_system_exists
+    @v3_only
+    @_verify_domain_control
+    async def set_exit_delay_away(call):
+        """Set the exit delay duration ("away" mode)."""
+        system = systems[call.data[ATTR_SYSTEM_ID]]
+        await system.set_exit_delay_away(call.data[ATTR_SECONDS])
+
+    @verify_system_exists
+    @v3_only
+    @_verify_domain_control
+    async def set_exit_delay_home(call):
+        """Set the exit delay duration ("home" mode)."""
+        system = systems[call.data[ATTR_SYSTEM_ID]]
+        await system.set_exit_delay_home(call.data[ATTR_SECONDS])
+
+    @verify_system_exists
+    @v3_only
+    @_verify_domain_control
+    async def set_light(call):
+        """Turn the base station light on/off."""
+        system = systems[call.data[ATTR_SYSTEM_ID]]
+        await system.set_light(call.data[ATTR_LIGHT_STATE])
+
+    @verify_system_exists
     @_verify_domain_control
     async def set_pin(call):
         """Set a PIN."""
-        system = systems[int(call.data[ATTR_SYSTEM_ID])]
+        system = systems[call.data[ATTR_SYSTEM_ID]]
         await system.set_pin(call.data[ATTR_PIN_LABEL], call.data[ATTR_PIN_VALUE])
 
+    @verify_system_exists
+    @v3_only
+    @_verify_domain_control
+    async def set_voice_prompt_volume(call):
+        """Set the volume of the door chime."""
+        await set_volume_parameter(call, "set_voice_prompt_volume")
+
     for service, method, schema in [
         ("remove_pin", remove_pin, SERVICE_REMOVE_PIN_SCHEMA),
+        ("set_alarm_duration", set_alarm_duration, SERVICE_SET_DURATION_SCHEMA),
+        ("set_alarm_volume", set_alarm_volume, SERVICE_SET_VOLUME_SCHEMA),
+        ("set_chime_volume", set_chime_volume, SERVICE_SET_VOLUME_SCHEMA),
+        ("set_entry_delay_away", set_entry_delay_away, SERVICE_SET_DURATION_SCHEMA),
```

Comment:
Why not have a single delay update service ?

Reply:
Totally fair. By that logic, I should probably consolidate the volume-related services, too, yes?

Reply:
Probably better yeah